src/main/java/org/opensearch/jobscheduler/rest/action/RestReleaseLockAction.java

```diff
@@ -0,0 +1,140 @@
+import java.util.Locale;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionListener;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.jobscheduler.JobSchedulerPlugin;
+import org.opensearch.jobscheduler.spi.LockModel;
+import org.opensearch.jobscheduler.spi.utils.LockService;
+import org.opensearch.jobscheduler.utils.JobDetailsService;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestRequest;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+import org.opensearch.rest.RestStatus;
+
+public class RestReleaseLockAction extends BaseRestHandler {
+
+    public static final String RELEASE_LOCK_ACTION = "release_lock_action";
+    public static final String LOCK_ID = "lock_id";
+
+    private final Logger logger = LogManager.getLogger(RestReleaseLockAction.class);
+
+    private LockService lockService;
+
+    private LockModel releaseLock;
+
+    private boolean releaseResponse;
+
+    public RestReleaseLockAction(LockService lockService) {
+        this.lockService = lockService;
+    }
+
+    @Override
+    public String getName() {
+        return RELEASE_LOCK_ACTION;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return ImmutableList.of(
+            new Route(PUT, String.format(Locale.ROOT, "%s/%s/{%s}", JobSchedulerPlugin.JS_BASE_URI, "_release_lock", LOCK_ID))
+        );
+    }
+
+    @Override
+    public RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient nodeClient) throws IOException {
+        String lockId = restRequest.param(LOCK_ID);
+        if (lockId == null || lockId.isEmpty()) {
+            throw new IOException("lockId cannot be null or empty");
+        }
+
+        CompletableFuture<LockModel> findInProgressFuture = new CompletableFuture<>();
+        lockService.findLock(lockId, ActionListener.wrap(lock -> {
+            releaseLock = lock;
+            findInProgressFuture.complete(lock);
+        }, exception -> {
+            logger.error("Could not find lock model with lockId " + lockId, exception);
+            findInProgressFuture.completeExceptionally(exception);
+        }));
+
+        try {
+            findInProgressFuture.orTimeout(JobDetailsService.TIME_OUT_FOR_REQUEST, TimeUnit.SECONDS).join();
+        } catch (CompletionException e) {
+            if (e.getCause() instanceof TimeoutException) {
+                logger.info(" Request timed out with an exception ", e);
+            } else {
+                throw e;
+            }
+        } catch (Exception e) {
+            logger.info(" Could not find lock model with lockId due to exception ", e);
```

Comment:
Have this log at debug level?[1] Info seems like a lot.
 

[1] https://www.tutorialspoint.com/log4j/log4j_logging_levels.htm

Reply:
ACk